core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -191,6 +203,26 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
   }
 }
 
+class FailedPartitions {
```

Comment:
A brief doc for this class may be helpful. I think it's worth mentioning that this includes partitions which failed either during truncation or appending. The basic three classes of errors are:
1) storage errors
2) fenced epoch
3) unexpected errors
We can mention that partitions which fail due to a storage error are eventually removed from this set after the log directory is taken offline.

Reply:
I have added doc for this class now.